packages/core/strapi/lib/services/entity-validator/index.js

```diff
@@ -222,10 +223,117 @@ const createValidateEntity =
         entity,
       },
       { isDraft }
-    ).required();
+    )
+      .test('relations-test', 'check that all relations exist', async function (data) {
+        try {
+          await checkRelationsExist(buildRelationsStore(model.uid, data) || {});
+        } catch (e) {
+          return this.createError({
+            path: this.path,
+            message: e.message,
+          });
+        }
+        return true;
+      })
+      .required();
+
     return validateYupSchema(validator, { strict: false, abortEarly: false })(data);
   };
 
+/**
+ * Builds an object containing all the media and relations being associated with an entity
+ * @param {String} uid of the model
+ * @param {Object} data
+ * @param {Object} relationsStore to be updated and returned
+ * @returns
+ */
+const buildRelationsStore = (uid, data, relationsStore = {}) => {
+  const currentModel = strapi.getModel(uid);
+
+  Object.keys(currentModel?.attributes || {}).forEach((attributeName) => {
```

Comment:
The model should exist and it should have attributes. If it's not the case then it means there is a problem somewhere else. In order to know that it should fail here so I think we should not be too flexible here. Also it's shorter and easier to read. Wdyt?
```suggestion
  Object.keys(currentModel.attributes).forEach((attributeName) => {
```

Reply:
👍 I think the problem was with the `dynamiczone` logic and has now been fixed with

```
return value.forEach((dzValue) => {
          if (!dzValue.__component) {
            return;
          }
          buildRelationsStore(dzValue.__component, dzValue, relationsStore);
});
```

WDYT?